app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCaptureRepositoryImpl.java

```diff
@@ -223,7 +223,12 @@ public ValidationStrategy validationStrategy() {
     @Override
     @Nullable
     public String getEnrollmentUid() {
-        return getCurrentEvent().enrollment();
+        Event currentEvent = getCurrentEvent();
+        if (currentEvent.enrollment() == null) {
```

Comment:
Should not be currentEvent == null? Calling currentEvent.enrollment() will throw nullpointer if currentEvent is null